build/base/src/extensions/option.rs

```diff
@@ -5,6 +5,18 @@
 // use crate::prelude::*;
 
 /// Extension methods for `Option`.
-pub trait OptionExt<T> {}
+pub trait OptionExt<T> {
+    /// Checks if the option is `Some` and contains the given value.
+    fn contains<U>(&self, x: &U) -> bool
+    where U: PartialEq<T>;
+}
 
-impl<T> OptionExt<T> for Option<T> {}
+impl<T> OptionExt<T> for Option<T> {
+    fn contains<U>(&self, x: &U) -> bool
+    where U: PartialEq<T> {
+        match self {
+            Some(y) => x.eq(y),
+            None => false,
+        }
+    }
```

Comment:
lints some ago recommended `is_some_and`.

Reply:
I find `is_some_and(|x| x == target)` less readable and somewhat annoying. What's worse, it takes ownership of the inner value, so it is not really a replacement in many cases. I think the recommendation was using `iter().any(…)` but I'd rather keep `contains`.